### PR TITLE
avistaz, cinemaz, privatehd: add imdb search and banner

### PR DESCRIPTION
**Pull request by ngosang**
* Add IMDB search
* Add banner / poster

Tested in privatehd only.

Part of #4859

### CODE REVIEW

**Comment by ngosang** [comment]
@RoloSoze you will have to test the other Avistaz trackers after the release. I don't have account.

**Comment by RoloSoze** [comment]
> 
> 
> @RoloSoze you will have to test the other Avistaz trackers after the release. I don't have account.

I'll do that. Thank you.

File: src/Jackett.Common/Indexers/Abstract/AvistazTracker.cs [hunk 1]
```diff
@@ -81,16 +85,25 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
-            var category = "0"; // Aka all
-            if (categoryMapping.Count() == 1)
+            var categoryMapping = MapTorznabCapsToTrackers(query).Distinct().ToList();
+            var qc = new NameValueCollection
             {
-                category = categoryMapping.First();
-            }
-
+                {"in", "1"},
+                {"type", categoryMapping.Any() ? categoryMapping.First() : "0"} // type=0 => all categories
```
Comment (cadatoiva) [inline src/Jackett.Common/Indexers/Abstract/AvistazTracker.cs]: This should be
`{"type", categoryMapping.FirstOrDefault() ?? "0"}`

File: src/Jackett.Common/Indexers/Abstract/AvistazTracker.cs [hunk 2]
```diff
@@ -81,16 +85,25 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
         {
             var releases = new List<ReleaseInfo>();
 
-            var categoryMapping = MapTorznabCapsToTrackers(query).Distinct();
-            var category = "0"; // Aka all
-            if (categoryMapping.Count() == 1)
+            var categoryMapping = MapTorznabCapsToTrackers(query).Distinct().ToList();
```
Comment (cadatoiva) [inline src/Jackett.Common/Indexers/Abstract/AvistazTracker.cs]: Changing to `.FirstOrDefault()` removes multiple enumeration. We can drop `.ToList()` here for better performance.

File: src/Jackett.Common/Indexers/Abstract/AvistazTracker.cs [hunk 3]
```diff
@@ -163,5 +179,27 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
+                    imdbResponse = await RequestStringWithCookiesAndRetry(imdbUrl, null, null, imdbHeaders);
+                }
+                var json = JsonConvert.DeserializeObject<dynamic>(imdbResponse.Content);
+                return (string)((JArray)json["data"])[0]["id"];
+            }
+            catch (Exception)
+            {
+                return null;
+            }
+        }
```
Comment (cadatoiva) [inline src/Jackett.Common/Indexers/Abstract/AvistazTracker.cs]: This function feels like it should go somewhere outside the tracker for others to use.

Comment (ngosang) [reply src/Jackett.Common/Indexers/Abstract/AvistazTracker.cs]: This function is used only by this tracker.

**Comment by ngosang** [comment]
@cadatoiva I'm not a C# developer so I do the best I know. I will follow your advice in the future.
My goal is not to write perfect code, is to solve the open issues and add more features to Jackett. The changes you requested are cosmetic / syntactic sugar to me and IMHO they are not enough reason to block the PRs. I prefer to spend my time in fixing more bugs.

**Pull request merged**